### PR TITLE
[RLlib] DQN rainbow eager-mode (keras style NoisyLayer) (preparation for native tf2.x support).

### PR DESCRIPTION
**Pull request by sven1977**
This PR moves the tf noisy layer from DQN into its own keras-style Layer class in preparation of supporting native tf2.x (with DQN).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/agents/dqn/distributional_q_tf_model.py [hunk 1]
```diff
@@ -220,20 +185,20 @@ def get_state_value(self, model_out):
 
         return self.state_value_head(model_out)
 
-    def _noisy_layer(self,
+    """def _noisy_layer(self,
```
Comment (ericl) [inline rllib/agents/dqn/distributional_q_tf_model.py:188]: Can we remove this?

File: ci/travis/ci.sh [hunk 2]
```diff
@@ -200,7 +200,7 @@ build_sphinx_docs() {
     if [ "${OSTYPE}" = msys ]; then
       echo "WARNING: Documentation not built on Windows due to currently-unresolved issues"
     else
-      sphinx-build -q -E -T -b html source _build/html
+      sphinx-build -q -E -W -T -b html source _build/html
```
Comment (ericl) [inline ci/travis/ci.sh:203]: What does the -W do?

**Review by ericl — APPROVED** [review]

File: ci/travis/ci.sh (hunk 2 above)
Comment (sven1977) [reply ci/travis/ci.sh:203]: Treat sphinx warnings as errors.

File: rllib/agents/dqn/distributional_q_tf_model.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/dqn/distributional_q_tf_model.py:188]: +1

Comment (sven1977) [reply rllib/agents/dqn/distributional_q_tf_model.py]: done

**Pull request merged**